src/views/Picklist-Detail.vue

```diff
@@ -143,69 +139,31 @@ export default defineComponent({
         picklist.isChecked = true;
       })
     },
-    async presentAlertConfirm(header, message) {
-      const alert = await alertController
-      .create({
-        header: header,
-        message: message,
-        buttons: [
-          {
-            text: this.$t('Okay')
-          },
-        ],
-      });
-      return alert.present();
-    },
-
-    // Native barcode scanner
-    async checkCameraPermission() {
-      const status = await BarcodeScanner.checkPermission({ force: true });
-      if (status.granted) {
-        // the user granted permission
-        return true;
-      } else if (status.denied) {
-        // the user has not given permission, showing alert message to enable camera permission in settings
-        this.presentAlertConfirm(translate('No permission'), translate('Please allow camera access in your settings'));
-      } else {
-        // showing alert if there is any other error
-        this.presentAlertConfirm(translate('Error'), translate('Unable to start camera, please try again'));
-      }
-      return false;
-    },
-    async startScan() {
-      const result = await BarcodeScanner.startScan(); // start scanning and wait for a result
-      // if the result has content
-      if (result.hasContent) {
-        BarcodeScanner.startScan({ targetedFormats: ['UPC_A'] })
-        this.scannerActive = false;
-        this.scanResult = result.content;
-        const item = this.picklistItem.find((product) => {
-          if (!product.isChecked) {
-            return product.productId === this.scanResult
+    async scanCode() {
+      const modal = await modalController
+        .create({
+          component: Scanner,
+        });
+      modal.onDidDismiss()
+        .then((result) => {
+          //result : value of the scanned barcode/QRcode
+          console.log(result);
+          const item = this.picklistItem.pickingItemList.find((product) => {
+            if (!product.isChecked) {
+              return product.productId === result.role
+            }
+          })
+          if (item) {
+            item.isChecked = true;
+          } else if (result.role && !item) {
+            showToast(translate("Product not found"))
           }
-        })
-        if (item) {
-          item.isChecked= true;
-        } else {
-          showToast(translate("Product not found"))
-        }
-      }
-    },
-    async stopScan() {
-      this.scannerActive = false;
-      await BarcodeScanner.stopScan();
-    },
-    async scan() {
-      const permissionGranted = await this.checkCameraPermission();
-      if(permissionGranted) {
-        this.scannerActive = true;
-        this.startScan();
-      } else {
-        this.stopScan();
-      }
+          result.role = "";
```

Comment:
Why this is needed?

Reply:
If we are not again assigning it to an empty link, then it takes the previous value and does not work on the second scan.

Reply:
I think this should be handled in generic scanner modal. On decode completion we could return data or error and reset the value otherwise this needs to be handled everytime we use it 
WDYT?

Reply:
Ok, let me check on this.